### PR TITLE
Critical! Fix test cases discoverage in Xcode 13.3

### PR DESCRIPTION
**Pull request by BobCatC**
Fix for #1123 

# Why critical?

Because Quick does not work with Xcode 13.3.
 
## What behavior was changed?

When running tests in Xcode 13.3 notification `testBundleWillStart` is called after method `testInvocations` of each QuickTest subclass. 
This means that when Xcode asks your QuickTest subclass for test cases to run (calling `testInvocations`) it gets empty array because test cases array will be computed on `testBundleWillStart` call.

## Step 2

As suggested @jessesquires this is step 1 with most important changes that fix the problem. Step 2 will contain a bit of refactoring with extracting examples loading into separate class as now it looks a bit strange that `QuickSpec.testInvocations` method calls `QuickSpecObservation`.

### CODE REVIEW

**Comment by kirhgoff** [comment]
Any news, dear collaborators?  Our big project has 0 tests executed, we are dreading the idea of rewriting the whole suite back to XCTest 😨

**Comment by jessesquires** [comment]
> we are dreading the idea of rewriting the whole suite back to XCTest 

It would be much easier to fork Quick, apply this patch, then point CocoaPods to your fork. 😊

**Review by vnntsu — APPROVED** [review]

**Comment by kirhgoff** [comment]
> > we are dreading the idea of rewriting the whole suite back to XCTest
> 
> It would be much easier to fork Quick, apply this patch, then point CocoaPods to your fork. 😊

We did something like that in the past, it requires a lot of effort of maintaining the forked repo and merging all the updates from the original one, and watching that they do not diverge. Easier to change the library

Any understanding why this is not being addressed - looks like code changes are there, what needs to be done to push it further?

**Comment by rbhuvizeli** [comment]
> Any news, dear collaborators? Our big project has 0 tests executed, we are dreading the idea of rewriting the whole suite back to XCTest 😨

Same here. Dear collaborators, pls. do sg.

**Comment by tsapeta** [comment]
@ikesyo sorry for pinging you, but this issue is very critical for many projects, could you please take a look at this PR and release a new version?

**Review by ashfurrow — APPROVED** [review]
Hello 👋 I've read up and understand the situation. These changes make sense to me.

I've been inactive in the Quick project for several years so core maintainers can let me know if I'm stepping out of line here. 

I have approved CI to run these changes and I'm approving the pull request. Once everything is green, we can merge this in and you can point CocoaPods to the `main` branch while we wait for the further refactors 👍 

In the PR conversation, it was suggested that contributors can point their `Podfile` to their fork of Quick project that already includes this fix. This is a good idea while you're waiting for this PR to be merged. I know it is a little extra work to do this, but you only need to wait as long as the PR remains open. You don't need to pull in upstream changes (provided that the PR gets merged relatively quickly – no pun intended). I highly recommend you do this to unblock your work, in lieu of pinging maintainers directly since we are volunteers, after all.

**Comment by ashfurrow** [comment]
The Danger failure looks like it was caused by configuration that doesn't pass the Danger secret to fork runs. This is common and we can probably ignore it.

**Comment by ashfurrow** [comment]
I ran Danger locally. Its only warning was of a missing unit test to accompany the `.swift` file changes. I will trust you to follow-up with that in your follow-up pull request.

I'm comfortable merging the now. Let's give this a shot...

**Comment by ashfurrow** [comment]
I had to temporarily remove Danger from the list of required checks to merge this pull request. It looks like it was failing to post the warning, but as I said earlier, it passed locally.

It doesn't look like Quick currently has many active contributors who work outside the project. I'm happy to help this one time, but I'm uncomfortable stepping outside the boundaries that the project maintainers have set up.

Good luck with your subsequent PRs, I hope that this resolves your immediate issue. And thank you for contributing to the project 🙇

**Comment by jessesquires** [comment]
Thank you so much @ashfurrow! 🙌🏼 You are appreciated my friend! 😄

**Comment by SD10** [comment]
Thank you to all the Quick maintainers and thanks Ash for pushing this through ❤️

**Comment by donnie-jp** [comment]
@ashfurrow Thanks for getting the fix merged!

Would you and other maintainers (@ikesyo ?) be supportive of this project being moved to MobileNativeFoundation so it can be more easily supported by the community - here's their [proposed adoption criteria](https://github.com/MobileNativeFoundation/discussions/discussions/46).

**Comment by tropicalappspoland** [comment]
New Quick release incoming?

**Comment by mauronate** [comment]
Hi! Would it be possible to restore this deleted branch? We were actually using it to build Quick directly from here, instead of using the official Quick release, by making cocoapods point at this branch when installing it, since no new release has been made so far, and this was the only way to use Quick on Xcode 13.3

**Comment by tsapeta** [comment]
@mauronate now you can point to `a0a5fc857cea079fbe973e4faa80b6ceaf17bd46` commit from the main branch 😉

**Pull request merged**
